### PR TITLE
Bill/output serial port

### PR DESCRIPTION
**Pull request by bill-luu**

### CODE REVIEW

File: CommunicationLayer/CommDeviceControl/OutputRadioCommDevice.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+OutputRadioCommDevice::OutputRadioCommDevice(I_CommDevice& inputSerialPort, QSerialPort& outputSerialPort, I_Settings& settings)
+    : outputSerialPort_(outputSerialPort)
+    , inputSerialPort_(inputSerialPort)
+{
+    setSerialParameters(settings.outputSerialPortName(), settings.outputBaudrate());
+
+    // This makes sure that it is connected after everything is created.
+    // Only if the config file enables it.
+    if (settings.outputSerialEnabled())
```
Comment (ExtraGravity) [inline CommunicationLayer/CommDeviceControl/OutputRadioCommDevice.cpp]: to not raise erreors, i would just check this at the beginning and then return if false.

**Review by ExtraGravity — APPROVED** [review]
Not janky at all, very good :) 

Just the one comment, but it's not that important. You can change it if you like.

File: CommunicationLayer/CommDeviceControl/OutputRadioCommDevice.cpp (hunk 1 above)
Comment (bill-luu) [reply CommunicationLayer/CommDeviceControl/OutputRadioCommDevice.cpp]: So then something like this?
```
    // This makes sure that it is connected after everything is created.
    // Only if the config file enables it.
    if (!settings.outputSerialEnabled())
    {
        return;
    }
    setSerialParameters(settings.outputSerialPortName(), settings.outputBaudrate());
    QTimer::singleShot(0, this, SLOT(connectToDataSource()));
    connect(&inputSerialPort_, SIGNAL(dataReceived(QByteArray)),
            this, SLOT(handleSerialDataIncoming(QByteArray)));
    connect(this, SIGNAL(dataReceived(QByteArray)),
            this, SLOT(forwardSerialData(QByteArray)));
```
Out of curiosity, how would that affect raising errors?

Comment (ExtraGravity) [reply CommunicationLayer/CommDeviceControl/OutputRadioCommDevice.cpp]: To make sure you actually do nothing if it's not enabled.

Who guarantees that `setSerialParameters(settings.outputSerialPortName(), settings.outputBaudrate());` doesn't throw any errors? Maybe it doesn't, but whoever is looking at this next will have to read it. Maybe it does, and we could find ourselves with a difficult bug.

**Pull request merged**